src/RA_md5factory.h

```diff
@@ -2,10 +2,11 @@
 #define RA_MD5FACTORY_H
 #pragma once
 
-#include "RA_Defs.h"
+#include "ra_fwd.h"
+#include <vector>
 
-extern std::string RAGenerateMD5(const std::string& sStringToMD5);
-extern std::string RAGenerateMD5(const BYTE* pIn, size_t nLen);
-extern std::string RAGenerateMD5(const std::vector<BYTE> DataIn);
+_NODISCARD std::string RAGenerateMD5(_In_ const std::string& sStringToMD5) noexcept;
+_NODISCARD std::string RAGenerateMD5(_In_ const BYTE* const pIn, _In_ size_t nLen) noexcept;
```

Comment:
Not really sure in C but the placement of const in C++ does have a difference. This makes `pIn `a "const pointer to const", meaning that the pointer and the object are immutable.